impl/src/main/java/io/jsonwebtoken/impl/compression/DefaultCompressionCodecResolver.java

```diff
@@ -54,19 +76,23 @@ public CompressionCodec resolveCompressionCodec(Header header) {
         if (!hasCompressionAlgorithm) {
             return null;
         }
-        if (io.jsonwebtoken.CompressionCodecs.DEFLATE.getAlgorithmName().equalsIgnoreCase(cmpAlg)) {
-            return io.jsonwebtoken.CompressionCodecs.DEFLATE;
-        }
-        if (io.jsonwebtoken.CompressionCodecs.GZIP.getAlgorithmName().equalsIgnoreCase(cmpAlg)) {
-            return io.jsonwebtoken.CompressionCodecs.GZIP;
-        }
-
-        throw new CompressionException("Unsupported compression algorithm '" + cmpAlg + "'");
+        return byName(cmpAlg);
     }
 
     private String getAlgorithmFromHeader(Header header) {
         Assert.notNull(header, "header cannot be null.");
 
         return header.getCompressionAlgorithm();
     }
+
+    private CompressionCodec byName(String name) {
+        Assert.hasText("'name' must not be empty");
+
+        CompressionCodec codec = codecs.get(name.toUpperCase());
+        if (codec == null) {
+            throw new CompressionException(String.format(MISSING_COMPRESSION_MESSAGE, name));
```

Comment:
I don't know if it's valid to use a CompressionException here, is it?  I mean, no compression/decompression is occurring - it's just a lookup failure.  Should this be something else?

Reply:
I had mixed feelings about that one too, in the end, I just kept the functionality the same (the previous version would throw a `CompressionException` if the codec could not be found.